src/player/Marine.cpp

```diff
@@ -72,3 +72,120 @@ int32_t Marine::checkForPickUp() {
+*
+* Function Interface: void Marine::updateImageWalk(double frameCount)
+*       double frameCount - counted frames while walking
+*
+* Description:
+*       This function repeatedly updates the image of the marine in order to animate 
+*       walking. It is called from GameStateMatch::handle after every frame and updates
+*       the feet placement of the marine.
+*/
+void Marine::updateImageWalk(const Uint8 *state, double frameCount) {
+    if (state[SDL_SCANCODE_UP] || state[SDL_SCANCODE_W]) {
+        if (getSpritePosX() == 0) {
+            setSpritePosX(SPRITE_SIZE_X);   //stops lag when taking first step
+        }
+        else if ((int)frameCount % FRAME_COUNT_WALK == 0) {
+            if (getSpritePosX() < SPRITE_SIZE_X*2) {
+                setSpritePosX(getSpritePosX() + SPRITE_SIZE_X); //moves to last walking position on sprite
+            } else {
+                setSpritePosX(SPRITE_SIZE_X);   //moves to first walking position on sprite
+            }
+        } 
+    }
+    else if (state[SDL_SCANCODE_DOWN] || state[SDL_SCANCODE_S]) {
+        if (getSpritePosX() == 0) {
+            setSpritePosX(SPRITE_SIZE_X);   //stops lag when taking first step
+        }
+        else if ((int)frameCount % FRAME_COUNT_WALK == 0) {
+            if (getSpritePosX() < SPRITE_SIZE_X*2) {
+                setSpritePosX(getSpritePosX() + SPRITE_SIZE_X); //moves to last walking position on sprite
+            } else {
+                setSpritePosX(SPRITE_SIZE_X);   //moves to first walking position on sprite
+            }
+        } 
+    }
+    else if (state[SDL_SCANCODE_LEFT] || state[SDL_SCANCODE_A]) {
+        if (getSpritePosX() == 0) {
+            setSpritePosX(SPRITE_SIZE_X);   //stops lag when taking first step
+        }
+        else if ((int)frameCount % FRAME_COUNT_WALK == 0) {
+            if (getSpritePosX() < SPRITE_SIZE_X*2) {
+                setSpritePosX(getSpritePosX() + SPRITE_SIZE_X); //moves to last walking position on sprite
+            } else {
+                setSpritePosX(SPRITE_SIZE_X);   //moves to first walking position on sprite
+            }
+        } 
+    }
+    else if (state[SDL_SCANCODE_RIGHT] || state[SDL_SCANCODE_D]) {
+        if (getSpritePosX() == 0) {
+            setSpritePosX(SPRITE_SIZE_X);   //stops lag when taking first step
+        }
+        else if ((int)frameCount % FRAME_COUNT_WALK == 0) {
+            if (getSpritePosX() < SPRITE_SIZE_X*2) {
+                setSpritePosX(getSpritePosX() + SPRITE_SIZE_X); //moves to last walking position on sprite
+            } else {
+                setSpritePosX(SPRITE_SIZE_X);   //moves to first walking position on sprite
+            }
+        } 
+    }
+    else if (state[SDL_SCANCODE_H]) {  
+        setSpritePosX(SPRITE_SIZE_X*3); //testing hit state (change to when HP delta > 0)
+    }
+    else if (state[SDL_SCANCODE_J]) {
+        setSrcRect(SPRITE_SIZE_X*4, SPRITE_SIZE_Y*7, 110, 100); //testing dead state (change to when HP == 0)
```

Comment:
and magic nums here; comment says "testing" ... does that mean this will change?